{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.2",
    "cveMetadata": {
        "cveId": "CVE-2025-6075",
        "assignerOrgId": "28c92f92-d60d-412d-b760-e73465c3df22",
        "state": "PUBLISHED",
        "assignerShortName": "PSF",
        "dateReserved": "2025-06-13T15:05:20.139Z",
        "datePublished": "2025-10-31T16:41:34.983Z",
        "dateUpdated": "2025-10-31T17:12:40.996Z"
    },
    "containers": {
        "cna": {
            "affected": [
                {
                    "defaultStatus": "unaffected",
                    "product": "CPython",
                    "repo": "https://github.com/python/cpython",
                    "vendor": "Python Software Foundation",
                    "versions": [
                        {
                            "version": "0",
                            "lessThan": "3.15.0",
                            "status": "affected",
                            "versionType": "python"
                        }
                    ]
                }
            ],
            "cpeApplicability": [
                {
                    "nodes": [
                        {
                            "cpeMatch": [
                                {
                                    "criteria": "cpe:2.3:a:python_software_foundation:cpython:*:*:*:*:*:*:*:*",
                                    "versionEndIncluding": "3.14.0",
                                    "versionStartIncluding": "0",
                                    "vulnerable": true
                                }
                            ],
                            "negate": false,
                            "operator": "OR"
                        }
                    ],
                    "operator": "OR"
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "supportingMedia": [
                        {
                            "base64": false,
                            "type": "text/html",
                            "value": "<div>If the value passed to os.path.expandvars() is user-controlled a \nperformance degradation is possible when expanding environment \nvariables.</div>"
                        }
                    ],
                    "value": "If the value passed to os.path.expandvars() is user-controlled a \nperformance degradation is possible when expanding environment \nvariables."
                }
            ],
            "metrics": [
                {
                    "cvssV4_0": {
                        "Automatable": "NOT_DEFINED",
                        "Recovery": "NOT_DEFINED",
                        "Safety": "NOT_DEFINED",
                        "attackComplexity": "LOW",
                        "attackRequirements": "PRESENT",
                        "attackVector": "LOCAL",
                        "baseScore": 1.8,
                        "baseSeverity": "LOW",
                        "privilegesRequired": "HIGH",
                        "providerUrgency": "NOT_DEFINED",
                        "subAvailabilityImpact": "NONE",
                        "subConfidentialityImpact": "NONE",
                        "subIntegrityImpact": "NONE",
                        "userInteraction": "NONE",
                        "valueDensity": "NOT_DEFINED",
                        "vectorString": "CVSS:4.0/AV:L/AC:L/AT:P/PR:H/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N",
                        "version": "4.0",
                        "vulnAvailabilityImpact": "LOW",
                        "vulnConfidentialityImpact": "NONE",
                        "vulnIntegrityImpact": "NONE",
                        "vulnerabilityResponseEffort": "NOT_DEFINED"
                    },
                    "format": "CVSS",
                    "scenarios": [
                        {
                            "lang": "en",
                            "value": "GENERAL"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "28c92f92-d60d-412d-b760-e73465c3df22",
                "shortName": "PSF",
                "dateUpdated": "2025-10-31T17:12:40.996Z"
            },
            "references": [
                {
                    "tags": [
                        "issue-tracking"
                    ],
                    "url": "https://github.com/python/cpython/issues/136065"
                },
                {
                    "tags": [
                        "vendor-advisory"
                    ],
                    "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/IUP5QJ6D4KK6ULHOMPC7DPNKRYQTQNLA/"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/2e6150adccaaf5bd95d4c19dfd04a36e0b325d8c"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/631ba3407e3348ccd56ce5160c4fb2c5dc5f4d84"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/892747b4cf0f95ba8beb51c0d0658bfaa381ebca"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/9ab89c026aa9611c4b0b67c288b8303a480fe742"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/c8a5f3435c342964e0a432cc9fb448b7dbecd1ba"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/f029e8db626ddc6e3a3beea4eff511a71aaceb5c"
                }
            ],
            "source": {
                "discovery": "UNKNOWN"
            },
            "title": "Quadratic complexity in os.path.expandvars() with user-controlled template",
            "x_generator": {
                "engine": "Vulnogram 0.4.0"
            }
        }
    }
}